daemon/logger/logentries/logentries.go

```diff
@@ -50,8 +50,10 @@ func New(info logger.Info) (logger.Logger, error) {
 		return nil, errors.Wrap(err, "error connecting to logentries")
 	}
 	var lineOnly bool
-	if lineOnly, err = strconv.ParseBool(info.Config[lineonly]); err != nil {
-		return nil, errors.Wrap(err, "error parsing lineonly option")
+	if info.Config[lineonly] != "" {
+		if lineOnly, err = strconv.ParseBool(info.Config[lineonly]); err != nil {
+			return nil, errors.Wrap(err, "error parsing lineonly option")
+		}
```

Comment:
Can you add a minimum test-case for this? Could be as simple as;

```go
package logentries

import (
	"testing"

	"github.com/docker/docker/daemon/logger"
	"github.com/stretchr/testify/require"
)

func TestNewWithEmptyOptions(t *testing.T) {
	_, err := New(logger.Info{})
	require.NoError(t, err)
}
```

Reply:
Can I also suggest changing the two ifs into one:
```Go
if lineOnly, err = strconv.ParseBool(info.Config[lineonly]); info.Config[lineonly] != "" && err != nil {
    ...
}
```

Reply:
@ripcurld0 Sorry but don't agree as the (unnecessary) cast statement would be executed in any case this way.

Reply:
@thaJeztah You know I even have done this and this obviously worked but then I noticed that actual TCP connection to Logentries is done in New() as well. Logentries driver used is a third party with service URL hardcoded so in this case the whole driver has to be mocked to test this properly.